src/tensor/operators_logical.nim

```diff
@@ -17,6 +17,7 @@ import  ./data_structure,
         ./init_cpu
 
 # TODO tests
+# TODO Empty tensors boolean logic?
```

Comment:
This is also difficult. I'd say that:
1. boolean operations only make sense if `a`, `b` have the same size. That would, in the current implementation using `map2_inline` automatically lead to an error if `a` was empty and `b` wasn't. That's imo fine.

Aside: currently `map2_inline` and friends only raise exceptions when compiled with `boundChecks` of course, so the errors might still be hard to reason about without checks, but that can't be helped?

2. if both tensors are empty, the result is also an empty tensor.

For a second I got confused by boolean tensor operations, which return `bool` and not `Tensor[bool]` where one could argue in terms of set theory for two empty tensors (-> two empty sets), but this doesn't apply here.

Reply:
The checks should probably reenabled, at all times. That was at a time where I was too optimization-happy ;)